connectors/es/sink.py

```diff
@@ -660,29 +663,50 @@ async def prepare_content_index(self, index, *, mappings=None):
         exists = await self.client.indices.exists(
             index=index, expand_wildcards=expand_wildcards
         )
+
+        mappings = Mappings.default_text_fields_mappings(is_connectors_index=True)
+
         if exists:
+            # Update the index mappings if needed
             self._logger.debug(f"{index} exists")
-            response = await self.client.indices.get_mapping(
-                index=index, expand_wildcards=expand_wildcards
+            await self._ensure_content_index_mappings(index, mappings, expand_wildcards)
+        else:
+            # Create a new index
+            self._logger.info("Attempt to create a new index")
+            await self._create_content_index(
+                index=index, language_code=language_code, mappings=mappings
             )
-            existing_mappings = response[index].get("mappings", {})
-            if len(existing_mappings) == 0 and mappings:
-                self._logger.debug(
-                    "Index %s has no mappings or it's empty. Adding mappings...", index
-                )
-                await self.client.indices.put_mapping(
-                    index=index,
-                    dynamic=mappings.get("dynamic", False),
-                    dynamic_templates=mappings.get("dynamic_templates", []),
-                    properties=mappings.get("properties", {}),
-                    expand_wildcards=expand_wildcards,
-                )
-                self._logger.debug("Index %s mappings added", index)
-            else:
-                self._logger.debug("Index %s already has mappings. Skipping...", index)
-            return
+            self._logger.info(f"Index {index} has been successfully created.")
+
+        return
+
+    async def _ensure_content_index_mappings(self, index, mappings, expand_wildcards):
```

Comment:
Why the `expand_wildcards` is needed here? I think we always use the exact index name instead of wildcard here?

Reply:
I'm not aware of any use cases where `expand_wildcards` is required but this option has been used from the beginning. In this PR, I'm not changing the index mapping (only moved it to a separate method) so this is why `expand_wildcards` stays. 

This is a good question but not in this PR scope.